contracts/tokens/SecurityToken.sol

```diff
@@ -402,13 +474,14 @@ contract SecurityToken is ISecurityToken {
 
     /**
      * @dev Burn function used to burn the securityToken
-     * @param _value No. of token that get burned     
+     * @param _value No. of token that get burned
      */
     function burn(uint256 _value) checkGranularity(_value) public {
         adjustInvestorCount(msg.sender, address(0), _value);
         require(tokenBurner != address(0), "Token Burner contract address is not set yet");
         require(verifyTransfer(msg.sender, address(0), _value), "Transfer is not valid");
         require(_value <= balances[msg.sender], "Value should no be greater than the balance of msg.sender");
+        adjustTotalSupplyCheckpoints();
```

Comment:
Shouldn't `adjustBalanceCheckpoints` be called here as well (below the burning of tokens)? If not an account would have a different amount of tokens than what balanceOfAt will return.

Reply:
Agreed - added.